addons/stock/models/stock_move.py

```diff
@@ -940,7 +940,7 @@ def _push_apply(self):
 
     def _merge_moves_fields(self):
         """ This method will return a dict of stock move’s values that represent the values of all moves in `self` merged. """
-        merge_extra = self.env.context.get('merge_extra')
+        merge_extra = self[0].state != 'waiting' and self.env.context.get('merge_extra')
```

Comment:
in `merge_moves()`, I would reset the `merge_extra` key to `False` when calling `_merge_moves_fields` if there is not `merge_into` value. Because `merge_extra` is only used with a specific stock move in `merge_into`